### PR TITLE
Add sort query parameter to revisionAndJobInfo

### PR DESCRIPTION
**Pull request by fviale**

### CODE REVIEW

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java [hunk 1]
```diff
@@ -177,6 +177,12 @@
      *            fetch running jobs
      * @param finished
      *            fetch finished jobs
+     * @param sortParams
+     *            string containing sort directives. It must contain a comma separated list of sort parameters.
```
Comment (aminelouati) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java:181]: string containing sort directives -> string composed of sorted directives

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java [hunk 2]
```diff
@@ -252,17 +277,29 @@ public boolean checkJobPermissionMethod(String sessionId, String method, String
...
+                    logger.warn("Invalid sort parameter string, using default : " + sortParams);
+                    sortParameterList = DEFAULT_JOB_SORT_PARAMS;
+                }
+            }
 
             Page<JobInfo> page = s.getJobs(index,
                                            limit,
                                            new JobFilterCriteria(onlyUserJobs, pending, running, finished),
-                                           DEFAULT_JOB_SORT_PARAMS);
+                                           sortParameterList);
```
Comment (aminelouati) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:302]: sortParameterList -> orderedParameterList

**Review by aminelouati — APPROVED** [review]

**Review by tao-xinxiu — APPROVED** [review]

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java (hunk 2 above)
Comment (fviale) [reply rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:302]: orderedParameterList ? I don't understand. It is one parameter containing the list of sort directives.

Comment (fviale) [reply rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:302]: the type is
List<SortParameter<JobSortParameter>>

Comment (aminelouati) [reply rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:302]: OK

**Comment by amouhoub** [comment]
@fviale could you please provide more explanation about the sortstring param for the endpoint so that it can be displayed in the auto-generated rest documentation.

**Comment by fviale** [comment]
@amouhoub I did in SchedulerRestInterface.java it's not enough ?

**Review by amouhoub — APPROVED** [review]

**Pull request merged**